src/toil/jobStores/aws/jobStore.py

```diff
@@ -681,6 +687,13 @@ def bucket_creation_pending(e):
                             raise
                         else:
                             return None
+                    elif e.status == 301:
+                        # This is raised if the user attempts to get a bucket in a region outside
+                        # the specified one, if the specified one is not `us-east-1`.  The us-east-1
+                        # server allows a user to use buckets from any region.
+                        assert self.region != 'us-east-1'
```

Comment:
Yes, I would just remove the assertion. The assumption we operate on is: 

If a `HEAD bucket` request on a connection to region A results in a 301, the bucket is in a region other than A. If it succeeds (200) but a subsequent `GET location` request over the same connection returns a region other than A, the bucket is also not in A.

Important thing to keep in mind: location is the same as region, except for region 'us-east-1' where location is the empty string. You should use toil.jobStores.aws.utils.region_to_bucket_location and toil.jobStores.aws.utils.bucket_location_to_region for translating between the two if you need to.

If we want to have 100% coverage we should also test this with a connection to `us-east-1` and the bucket in one of the other two US regions.